// react-generate-unique-key-for-map/dist/main.js
var $8zHUo$react = require("react");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $a2814682fe9c93f1$exports = {};

$parcel$export($a2814682fe9c93f1$exports, "useGetUniqueKey", () => $a2814682fe9c93f1$export$e3ae60773e14b60f);

var $716d45a28ab8b323$exports = {};

$parcel$export($716d45a28ab8b323$exports, "generateUniqueKey", () => $716d45a28ab8b323$export$a2159b42596d12b9);
function $716d45a28ab8b323$export$a2159b42596d12b9() {
    function s4() {
        return Math.floor((1 + Math.random()) * 65536).toString(16).substring(1);
    }
    return s4() + s4() + "-" + s4() + "-" + s4() + "-" + s4() + "-" + s4() + s4() + s4();
}


function $a2814682fe9c93f1$export$e3ae60773e14b60f() {
    const uniqueKeysRef = $8zHUo$react.useRef(new Map());
    function getUniqueKey(item) {
        if (uniqueKeysRef.current.has(item)) return uniqueKeysRef.current.get(item);
        const uniqueKey = $716d45a28ab8b323$export$a2159b42596d12b9();
        uniqueKeysRef.current.set(item, uniqueKey);
        return uniqueKey;
    }
    return getUniqueKey;
}



$parcel$exportWildcard(module.exports, $a2814682fe9c93f1$exports);
$parcel$exportWildcard(module.exports, $716d45a28ab8b323$exports);
